tests/test-vectors.js

```diff
@@ -831,6 +831,16 @@ var generalVectors = [
 	id: 50,
 	input: "abc <!-- 123",
 	output: "abc "
+},
+{
+	id: 51,
+	input: "<img src=\"x\" id=\'\" onerror=\"alert(1)\' />",
+	output: "<img src=\"x\" id=\'\" onerror=\"alert(1)\' />"
+},
+{
+	id: 52,
+	input: "<a href='http://www.yahoo.com/' />hello",
+	output: "<a href='http://www.yahoo.com/'>hello</a>"
```

Comment:
@maditya, this is the example that triggered me to ignore self-closing solidus.

but then i encounter some html samples of optional tags, that *may* require extra balancing logics
```html
<select><option>123<option>abc</option></select>
<table><tr><td>123<tr><td>abc</td></tr></table>
```
our current balancing logic will produce:
```html
<select><option>123<option>abc</option></option></select>
<table><tr><td>123<tr><td>abc</td></tr></td></tr></table>
```
while it looked very ugly, as one properly expects the following, but it worked out to produce the same DOM in Chrome:
```html
<select><option>123</option><option>abc</option></select>
<table><tr><td>123</td></tr><tr><td>abc</td></tr></table>
```